Lib/typing.py

```diff
@@ -827,8 +824,8 @@ def __class_getitem__(cls, params):
     def __init_subclass__(cls, *args, **kwargs):
         tvars = []
         if ('__orig_bases__' in cls.__dict__ and Generic in cls.__orig_bases__ or
-                '__orig_bases__' not in cls.__dict__ and Generic in cls.__bases__
-                and cls.__name__ != '_Protocol'):
+                '__orig_bases__' not in cls.__dict__ and Generic in cls.__bases__ and
```

Comment:
Too long line.

The condition is too complex and contains the same test (`'__orig_bases__' in cls.__dict__`) twice. I would write it in the following form:
```python
if '__orig_bases__' in cls.__dict__:
    if Generic in cls.__orig_bases__:
        raise TypeError("Cannot inherit from plain Generic")
else:
    if Generic in cls.__bases__ and cls.__name__ != '_Protocol':
        raise TypeError("Cannot inherit from plain Generic")
```
or
```python
if '__orig_bases__' in cls.__dict__:
    tmp = Generic in cls.__orig_bases__
else:
    tmp = Generic in cls.__bases__ and cls.__name__ != '_Protocol'
if tmp:
    raise TypeError("Cannot inherit from plain Generic")
```